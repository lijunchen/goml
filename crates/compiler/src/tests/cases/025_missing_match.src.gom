package main

import (
    "fmt"
)

func unit_to_string(x struct{}) string {
    return "()"
}

func bool_to_string(x bool) string {
    if x {
        return "true"
    } else {
        return "false"
    }
}

func int_to_string(x int) string {
    return fmt.Sprintf("%d", x)
}

func int_add(x int, y int) int {
    return x + y
}

func int_sub(x int, y int) int {
    return x - y
}

func int_mul(x int, y int) int {
    return x * y
}

func int_div(x int, y int) int {
    return x / y
}

func int_less(x int, y int) bool {
    return x < y
}

func string_add(a string, b string) string {
    return a + b
}

func string_print(s string) struct{} {
    fmt.Print(s)
    return struct{}{}
}

func string_println(s string) struct{} {
    fmt.Println(s)
    return struct{}{}
}

func missing(s string) struct{} {
    println("missing: " + s)
    panic("")
    return struct{}{}
}

type Point struct {
    x int
    y int
}

func main0() struct{} {
    var ret3 struct{}
    var p0__0 Point = Point{
        x: 0,
        y: 0,
    }
    var mtmp0 Point = p0__0
    var x1 int = mtmp0.x
    var x2 int = mtmp0.y
    switch x2 {
    case 8:
        switch x1 {
        case 10:
            ret3 = struct{}{}
        default:
            ret3 = missing("")
        }
    default:
        ret3 = missing("")
    }
    return ret3
}

func main() {
    main0()
}
