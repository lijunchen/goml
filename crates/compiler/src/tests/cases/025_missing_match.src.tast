fn main() -> unit {
  let p0/0: Point = Point { x: 0, y: 0 } in
  let Point { x: 10, y: 8 } = (p0/0 : Point) in
  ()
}