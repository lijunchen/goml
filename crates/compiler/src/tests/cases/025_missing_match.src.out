missing: 
panic: 

goroutine 1 [running]:
main.missing(...)
	${WORKDIR}/main.go:51
main.main0(...)
	${WORKDIR}/main.go:78
main.main()
	${WORKDIR}/main.go:84 +0x65
exit status 2
